tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -151,101 +161,184 @@ int main(int argc, char** argv) {
+
+            dnsduration = (double)usduration;
+            dsduration = dnsduration / ((double)1000000000);
+            bpersec = (DATA_SIZE * NUM_KERNEL_DDR) / dsduration;
+            mbpersec = (2 * bpersec) / ((double)1024 * 1024); // For concurrent Read/Write
+
+            if (mbpersec > max_throughput) max_throughput = mbpersec;
         }
 
-        for (int i = 0; i < NUM_KERNEL; i++) {
-            OCL_CHECK(err, err = q.enqueueWriteBuffer(input_buffer[i], CL_TRUE, 0, vector_size_bytes, input_host.data(),
+        std::cout << "Overall DDRs (total " << NUM_KERNEL_DDR << ") Throughput: " << max_throughput << "MB/s\n";
+    }
+    if (chk_hbm_mem) {
+        for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
+            unsigned int DATA_SIZE = i;
+
+            if (xcl::is_emulation()) {
+                reps = 2;
+                if (DATA_SIZE > 8 * 1024) break;
+            }
+
+            unsigned int vector_size_bytes = DATA_SIZE;
+            std::vector<unsigned char, aligned_allocator<unsigned char> > input_host(DATA_SIZE);
+            std::vector<unsigned char, aligned_allocator<unsigned char> > output_host(DATA_SIZE);
+
+            for (uint32_t j = 0; j < DATA_SIZE; j++) {
+                input_host[j] = j % 256;
+            }
+
+            // Initializing output vectors to zero
+            std::fill(output_host.begin(), output_host.end(), 0);
+
+            cl::Buffer input_buffer, output_buffer;
+
+            // These commands will allocate memory on the FPGA. The cl::Buffer objects
+            // can
+            // be used to reference the memory locations on the device.
+            // Creating Buffers
+            OCL_CHECK(err, input_buffer = cl::Buffer(context, CL_MEM_READ_WRITE, vector_size_bytes, nullptr, &err));
+            OCL_CHECK(err, output_buffer = cl::Buffer(context, CL_MEM_READ_WRITE, vector_size_bytes, nullptr, &err));
+
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(0, input_buffer));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(1, output_buffer));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(2, DATA_SIZE));
+            OCL_CHECK(err, err = krnls[NUM_KERNEL - 1].setArg(3, reps));
+
+            OCL_CHECK(err, err = q.enqueueWriteBuffer(input_buffer, CL_TRUE, 0, vector_size_bytes, input_host.data(),
                                                       nullptr, nullptr));
             OCL_CHECK(err, err = q.finish());
-        }
 
-        std::chrono::high_resolution_clock::time_point timeStart;
-        std::chrono::high_resolution_clock::time_point timeEnd;
+            std::chrono::high_resolution_clock::time_point timeStart;
+            std::chrono::high_resolution_clock::time_point timeEnd;
 
-        timeStart = std::chrono::high_resolution_clock::now();
+            timeStart = std::chrono::high_resolution_clock::now();
```

Comment:
As earlier.  and don't mix and match camelCase with under_bar_names.   My preference is latter, but consistency is what is important.